mitodl/ccxcon
# Create Endpoint fixes for edX to work properly.

#### What's this PR do?

Updates the API to take into account missing information from the edx side of things.
#### Where should the reviewer start?

`courses/views.py`
#### How should this be manually tested?

It should be run against https://github.com/mitocw/edx-platform/pull/153
1. Download Giovanni's branch.
2. Edit your vm's `/etc/hosts` file to point to the ip of your docker file with the hostname `localhost.daplie.com`.
3. Add edx user on ccxcon. They should have an edx_instance (at the bottom of the form) that's valid.
4. Add an OAuth Application object in the CCXCon admin. You'll need the generated id and secret later. It should belong to the edx user you just made.
5. Create a CCXCon object in edX (using shell) with the URL of https://localhost.daplie.com:8077/ and id and secret tokens generated in the previous step.
6. Enable CCX for a course.
7. Put in the above daplie url (with port) in your CCXcon info.
8. Save settings.
9. You should see the resulting course in https://localhost.daplie.com:8077/api/v1/coursexs/

To get the IP of your docker image, run this:

`docker ps | grep "ccxcon_web_1" | awk '{ print $1 }' | xargs docker inspect | grep 'IPAddress"' | head -n 1 | awk '{print $2}' | tr '"' ' ' | tr "," " "`
#### Any background context you want to provide?

It's difficult to get the base url from edx, so we can't get the proper image path either. edX also doesn't know if its a create or update, so we just handle it all seamlessly.
#### What are the relevant tickets?

Fixes #70 
#### What gif best describes this PR or how it makes you feel?

![](https://media.giphy.com/media/mBm6k5XoUJABy/giphy.gif)
